command_line/to_shelxcde.py

```diff
@@ -1,54 +1,179 @@
                     "eof",
                     "",
                 ]
             )
         )
 
 
-if __name__ == "__main__":
-    if len(sys.argv) > 3:
-        sites = int(sys.argv[3])
-        to_shelxcde(sys.argv[1], sys.argv[2], sites)
+def run(args):
+    parser = argparse.ArgumentParser(formatter_class=argparse.RawDescriptionHelpFormatter,
+                                     description="xia2.to_shelxcde - Generate script"
+                                     " to run SHELXC on the input SAD/MAD datasets",
+                                     epilog="Following combinations of the input parameters are valid:"
+                                     "\n  sad, label - for SAD dataset. label parameter is optional if only one dataset is available."
+                                     "\n  peak, infl, hrem, lrem - for MAD datasets that are read from different files."
+                                     "\n  mad, peak, infl, hrem, lrem - for MAD datasets read from a single multi-wavelength file."
+                                     "\n                                If labels are not provided datasets will be labeled according to wavelength."
+                                     "\n  nat - native dataset can be provided with any of the previous parameter combinations.")
+    parser.add_argument("--peak", help="Peak MAD dataset file/column label")
+    parser.add_argument("--infl", help="Inflection MAD dataset file/column label")
+    parser.add_argument("--hrem", help="High-remote MAD dataset file/column label")
+    parser.add_argument("--lrem", help="Low-remote MAD dataset file/column label")
+    parser.add_argument("--label", help="Column label for selecting SAD dataset")
+    parser.add_argument("--sad", help="SAD dataset file")
+    parser.add_argument("--mad", help="MAD dataset file")
+    parser.add_argument("--nat", help="Native dataset file")
+    parser.add_argument("-s", "--sites", type=int, help="Number of atoms to search")
+    parser.add_argument("prefix", help="Output file name prefix")
+
+    vals = parser.parse_args(args)
+    params = {k: v for (k, v) in vars(vals).items() if v}
+    param_names = set(params.keys())
+
+    allowed_groups = (set(('peak', 'infl', 'hrem', 'lrem', 'nat', 'prefix', 'sites')),
+                      set(('sad', 'label', 'nat', 'prefix', 'sites')),
+                      set(('mad', 'peak', 'infl', 'hrem', 'lrem', 'nat', 'prefix', 'sites')))
+    if not any(param_names.issubset(grp) for grp in allowed_groups):
+        parser.error('Invalid combination of the input parameters')
+
+    if 'sad' in params:
+        shelx_files = export_sad_hkl(params)
+    elif 'mad' in params:
+        shelx_files = export_single_mad_hkl(params)
+    elif 'peak' in params or 'infl' in params or 'hrem' in params or 'lrem' in params:
+        shelx_files = export_mad_hkl(params)
     else:
-        to_shelxcde(sys.argv[1], sys.argv[2])
+        raise ValueError("Input dataset file not specified")
```

Comment:
since the function builds on `argparse` anyway this is probably more helpful to the user:
```suggestion
        parser.error("Input dataset file not specified")
```

(untested)